test/mesh_test.jl

```diff
@@ -0,0 +1,52 @@
+using Test
+using LinearAlgebra
+using Gmsh
+using Inti
+using StaticArrays
+
+@testset "Native mesh generation" begin
+    Inti.clear_entities!()
+    r = rx = ry = 0.5
+    f = let r = r
+        (x) -> SVector(r * cos(2π * x[1]), r * sin(2π * x[1]))
+    end
+    arc1 = Inti.parametric_curve(f, 0, 0.5)
+    arc2 = Inti.parametric_curve(f, 0.5, 1)
+    Γ = Inti.Domain(arc1, arc2)
+    msh = Inti.meshgen(Γ, (100,))
+    quad = Inti.Quadrature(msh; qorder = 2)
+    @test Inti.integrate(x -> 1, quad) ≈ 2 * π * r
+
+    # transfinite patch
+    Inti.clear_entities!()
+    l1 = Inti.parametric_curve(0, 1) do x
+        return SVector(x[1], 0.1 * sin(2π * x[1]))
+    end
+    l2 = Inti.line(SVector(1.0, 0.0), SVector(1.0, 1.0))
+    l3 = Inti.line(SVector(1.0, 1.0), SVector(0.0, 1.0))
+    l4 = Inti.line(SVector(0.0, 1.0), (0.0, 0.0))
+    sq = Inti.transfinite_square(l1, l2, l3, l4)
+    Ω = Inti.Domain(sq)
+    msh = Inti.meshgen(Ω, Dict(l1 => 10, l2 => 20))
+    Γ = Inti.external_boundary(Ω)
+    msh[Γ]
+    Γ_msh = Inti.meshgen(Γ, (100,))
+end
+
+@testset "Gmsh backend" begin
+    r = rx = ry = 0.5
+    Inti.clear_entities!()
+    gmsh.initialize()
+    gmsh.option.setNumber("General.Verbosity", 2)
+    gmsh.option.setNumber("Mesh.MeshSizeMax", 0.005)
+    Inti.clear_entities!()
+    gmsh.model.occ.addDisk(0, 0, 0, rx, ry)
+    gmsh.model.occ.synchronize()
+    gmsh.model.mesh.generate(2)
+    Ω, M = Inti.import_mesh_from_gmsh_model(; dim = 2)
+    gmsh.finalize()
+    Γ = Inti.external_boundary(Ω)
+    dict = Inti.topological_neighbors(M[Γ])
```

Comment:
This does not seem to have been implemented, or rather pushed to this PR.

Reply:
You are right, this is in some other brach related to the local dim methods. Do
you want it or can I leave it to be added later?

Reply:
But you spotted a problem: the `runtests.jl` did not run all tests. I will fix
it now.

Reply:
It doesn't seem essential to have it in this PR, and since it is logically distinct it can be added a bit later.